### PR TITLE
v5.12 changelog

### PR DESCRIPTION
**Pull request by amyblais**

### CODE REVIEW

File: source/administration/changelog.md [hunk 1]
```diff
@@ -4,6 +4,123 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+
+#### Command Line Interface (CLI)
+ - Added a ``command modify`` CLI command to be able to modify slash commands.
+ - Added support for converting user accounts to bot accounts through the CLI.
+ - Implemented a new command ``MigrateConfig`` for migrating configuration to and from the database.
+ - For LDAP Groups, added the ability to switch a team or channel to be group-constrained via the CLI.
+ 
+#### Administration
+ - Added support for running two Mattermost instances on the same domain using subpaths.
+ - Added support for importing threads from Slack.
```
Comment (jasonblais) [inline source/administration/changelog.md]: what about the corporate Slack import?

File: source/administration/changelog.md [hunk 2]
```diff
@@ -4,6 +4,123 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+Multiple setting options were added to `config.json`. Below is a list of the additions and their default values on install. The settings can be modified in `config.json`, or the System Console when available.
+
+#### Changes to Team Edition and Enterprise Edition:
+
+ - Under ``"PluginSettings":`` in ``config.json``:
+   - Added ``"EnableHealthCheck": true``, to ensure all plugins are periodically monitored, and restarted or deactivated based on their health status.
+ - Under ``"NotificationLogSettings":`` in ``config.json``:
+   - Added ``"EnableConsole": true``, ``"ConsoleLevel": "DEBUG"``, ``"ConsoleJson": true``, ``"EnableFile": true``, ``"FileLevel": "INFO"``, ``"FileJson": true``, and ``"FileLocation": ""``, to implement a structured logger to keep track of push notifications.
+ - Under ``"ServiceSettings":`` in ``config.json``:
+   - Added ``"EnableBotAccountCreation": false`` and ``"DisableBotsWhenOwnerIsDeactivated": true``, to add the ability to enable bot accounts.
```
Comment (jasonblais) [inline source/administration/changelog.md]:
```suggestion
   - Added ``"EnableBotAccountCreation": false`` to enable bot account creation.
   - Added ``"DisableBotsWhenOwnerIsDeactivated": true`` to disable bots automatically when the owner is deactivated.
```

File: source/administration/changelog.md [hunk 3]
```diff
@@ -4,6 +4,123 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+   - Added ``"TrustedProxyIPHeader": []``, to explicitly define which IP headers are trusted.
+
+### Database Changes
+ - 
+
+### API Changes
```
Comment (jasonblais) [inline source/administration/changelog.md]: 1) Propose breaking to "RESTful API Changes" and "Plugin API" changes (for plugins there is only one, the last one).

image

This helps make it more clear which APIs the changes refer to.

2) Propose formatting the API changes consistently, similar to the screenshot above. Let me know if I can help.

File: source/administration/changelog.md [hunk 4]
```diff
@@ -4,6 +4,123 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+ - LDAP Group Sync:
+    - Added API endpoints ``getGroupsByChannel`` and ``GetGroupsByTeam`` to retrieve groups by team and by channel.
+    - Added ``group_constrained`` API to both ``/users`` and ``/users/search`` endpoints to be able to limit users listed to those allowed by group-constraints.
+ - Added ``KVCompareAndSet(key string, old []byte, new []byte)`` to Plugin API to add support for transactional semantics with KV Store in plugin framework.
+
+### Websocket Event Changes
+ - 
+
+### Known Issues
+ - Buttons inside ephemeral posts are not clickable / functional on the mobile app.
```
Comment (jasonblais) [inline source/administration/changelog.md]:
```suggestion
 - Buttons inside ephemeral messages are not clickable / functional on the mobile app.
```

File: source/administration/changelog.md [hunk 5]
```diff
@@ -4,6 +4,123 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+ - On a server using a subpath, the URL opens a blank page if the system admin changes the Site URL in the System Console UI. The system admin should restart the server to fix it.
```
Comment (jasonblais) [inline source/administration/changelog.md]:
```suggestion
 - On a server using a subpath, the URL opens a blank page if the System Admin changes the Site URL in the System Console UI. To fix, the System Admin should restart the server.
```

File: source/administration/changelog.md (hunk 1 above)
Comment (amyblais) [reply source/administration/changelog.md]: The PR hasn't been merged yet. Should I add it already?

File: source/administration/changelog.md [hunk 6]
```diff
@@ -4,6 +4,133 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+   - Added ``"EnableBotAccountCreation": false`` to enable bot account creation.
+   - Added ``"DisableBotsWhenOwnerIsDeactivated": true`` to disable bots automatically when the owner is deactivated.
+   - Added ``"TrustedProxyIPHeader": []``, to explicitly define which IP headers are trusted.
+
+### Database Changes
+ - 
+
+### API Changes
+ - Updated API to use gziphandler wrapper if server is configured to use gzip. This ensures that the Mattermost server can respond to REST API requests with compressed data (via gzip) to reduce the amount of bandwidth used.
+ - Added an API command to create and delete a team and channel.
```
Comment (jasonblais) [inline source/administration/changelog.md]: This is actually for Cypress end-to-end testing only, not for interacting with Mattermost https://github.com/mattermost/mattermost-webapp/pull/2601#issuecomment-498448815

File: source/administration/changelog.md [hunk 7]
```diff
@@ -4,6 +4,133 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+ - Users no longer have to rely on creating fake user accounts to act as bots for integrations. Instead, create a real bot account and use it to generate bot access tokens to interact with users and complete tasks.
+ - Users can can also use these bots to post to any channel in the system, whether it’s a private team, private channel or a direct message channel.
+ - For Enterprise deployments, bot accounts no longer count as an active user towards licensing subscriptions.
+ 
+#### Jira Plugin 2.0
+ - Enhanced existing plugin for a deep two-way integration between Jira and Mattermost.
+ - Send notifications for Jira issue creation, issue updates and comments to Mattermost channels.
+ - Users can also take quick actions in Mattermost, including creating Jira issues, attaching Mattermost messages to Jira issues, and transitioning issues via slash commands.
+ - For a full feature set for 2.0, see https://github.com/mattermost/mattermost-plugin-jira#jira-20-features.
+
```
Comment (jasonblais) [inline source/administration/changelog.md]: As another highlight, we can include the list of new pre-packaged plugins we're shipping in 5.12. (Would be good to add to release announce as well)

```
#### Pre-packaged Plugins
 - New pre-packaged plugins bundled with this Mattermost release include:
   - [GitHub plugin](https://github.com/mattermost/mattermost-plugin-github) for notifications, reminders and slash commands to stay up-to-date on issues and pull requests. Supports GitHub SaaS and Enterprise versions.
   - [Autolink plugin](https://github.com/mattermost/mattermost-plugin-autolink) to automatically hyperlink text, such as adding links to your issue tracker when someone posts an issue key or number.
   - [Custom Attributes plugin](https://github.com/mattermost/mattermost-plugin-custom-attributes) to add custom attributes in the user profile popover.
   - [Welcome Bot plugin](https://github.com/mattermost/mattermost-plugin-welcomebot) to improve onboarding and HR processes by adding a Welcome Bot that helps add new team members to channels.
   - [Amazon SNS CloudWatch plugin](https://github.com/mattermost/mattermost-plugin-aws-sns) to send alert notifications from [Amazon AWS CloudWatch](https://aws.amazon.com/cloudwatch/) to Mattermost channels via AWS SNS.

File: source/administration/changelog.md (hunk 1 above)
Comment (jasonblais) [reply source/administration/changelog.md]: Ah, didn't realize it wasn't merged yet. I guess it's gonna be too late for 5.12. Are we doing a custom build for the customer instead?

File: source/administration/changelog.md [hunk 8]
```diff
@@ -4,6 +4,123 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+#### Plugins
+ - Added support for markdown in plugin system console help text fields.
+ - Added support for plugins to override ephemeral posts.
+
+#### Localization
+ - Promoted Polish language to "official".
+
+#### Command Line Interface (CLI)
+ - Added a ``command modify`` CLI command to be able to modify slash commands.
+ - Added support for converting user accounts to bot accounts through the CLI.
```
[the remark this answers is not in the document]
Comment (tgly307) [reply source/administration/changelog.md]: I found a similar command here [mattermost user convert --bot github](https://github.com/mattermost/mattermost-plugin-github/blob/master/README.md#step-3-configure-plugin-in-mattermost), is this the new command for user group?

[the remark this answers is not in the document]
Comment (jasonblais) [reply source/administration/changelog.md]: `mattermost user convert --bot` is indeed the new CLI command to convert user accounts to bots 👍

File: source/administration/changelog.md [hunk 9]
```diff
@@ -4,17 +4,158 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+ - Updated API to use gziphandler wrapper if server is configured to use gzip. This ensures that the Mattermost server can respond to REST API requests with compressed data (via gzip) to reduce the amount of bandwidth used.
+ - LDAP Group Sync:
+    - Added API endpoints ``getGroupsByChannel`` and ``GetGroupsByTeam`` to retrieve groups by team and by channel.
+    - Added ``group_constrained`` API to both ``/users`` and ``/users/search`` endpoints to be able to limit users listed to those allowed by group-constraints.
+    
+#### Plugin API v4 Changes
+ - Added ``KVCompareAndSet(key string, old []byte, new []byte)`` to Plugin API to add support for transactional semantics with KV Store in plugin framework.
+
+### Websocket Event Changes
+ -
```
Comment (jasonblais) [inline source/administration/changelog.md]: Maybe check with SET?

File: source/administration/changelog.md [hunk 10]
```diff
@@ -4,17 +4,158 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+
+### Database Changes
+ - 
+
+### API Changes
+
+#### RESTful API v4 Changes
+ - Updated API to use gziphandler wrapper if server is configured to use gzip. This ensures that the Mattermost server can respond to REST API requests with compressed data (via gzip) to reduce the amount of bandwidth used.
+ - LDAP Group Sync:
+    - Added API endpoints ``getGroupsByChannel`` and ``GetGroupsByTeam`` to retrieve groups by team and by channel.
```
Comment (jasonblais) [inline source/administration/changelog.md]: Propose asking Martin to confirm endpoints for LDAP group sync

File: source/administration/changelog.md [hunk 11]
```diff
@@ -4,17 +4,158 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+   - Added ``"EnableHealthCheck": true``, to ensure all plugins are periodically monitored, and restarted or deactivated based on their health status.
+ - Under ``"NotificationLogSettings":`` in ``config.json``:
+   - Added ``"EnableConsole": true``, ``"ConsoleLevel": "DEBUG"``, ``"ConsoleJson": true``, ``"EnableFile": true``, ``"FileLevel": "INFO"``, ``"FileJson": true``, and ``"FileLocation": ""``, to implement a structured logger to keep track of push notifications.
+ - Under ``"ServiceSettings":`` in ``config.json``:
+   - Added ``"EnableBotAccountCreation": false`` to enable bot account creation.
+   - Added ``"DisableBotsWhenOwnerIsDeactivated": true`` to disable bots automatically when the owner is deactivated.
+   - Added ``"TrustedProxyIPHeader": []``, to explicitly define which IP headers are trusted.
+
+### Database Changes
+ -
```
Comment (jasonblais) [inline source/administration/changelog.md]:
```
`SchemeGuest` column added to the `TeamMembers` table.
`SchemeGuest` column added to the `ChannelMembers` table.
`DefaultTeamGuestRole` column added to the `Schemes` table, and set to an empty string.
`DefaultChannelGuestRole` column added to the `Schemes` table, and set to an empty string.
```

(from https://github.com/mattermost/mattermost-server/blob/master/store/sqlstore/upgrade.go#L683)

@jespino may be able to confirm if the above is correct

Comment (jespino) [reply source/administration/changelog.md]: Yes, the @jasonblais list of changes is correct, I added that fields as part of the guest account preparation changes.

File: source/administration/changelog.md (hunk 10 above)
Comment (amyblais) [reply source/administration/changelog.md]: @skiplist ^

File: source/administration/changelog.md (hunk 9 above)
Comment (amyblais) [reply source/administration/changelog.md]: @enahum / @jespino Were there any Websocket changes in 5.12?

Comment (enahum) [reply source/administration/changelog.md]: Not that I’m aware of

File: source/administration/changelog.md (hunk 10 above)
Comment (mkraft) [reply source/administration/changelog.md]: Also new APIs for...

* List/search groups
* Regenerate team invite ids
* Invalidate team email invites
* `/notifications/ack` (something to do with push notifications)

Let me know if you need more detail.

**Pull request merged**